test/src/auxkernels/CoupledAux.C

```diff
@@ -10,9 +10,11 @@
 #include "CoupledAux.h"
 
 registerMooseObject("MooseTestApp", CoupledAux);
+registerMooseObject("MooseTestApp", ADCoupledAux);
 
+template <bool is_ad>
 InputParameters
-CoupledAux::validParams()
+CoupledAuxTempl<is_ad>::validParams()
```

Comment:
It seems like these changes are really only for testing purposes? Is there any reason in production to use an `ADCoupledAux`?

Reply:
And I see that you are modifying a test material ... so do we really need these aux kernel changes?

Reply:
I thought the changes didn't overlap but they might have ended up overlapping. I ll check 

No reason to use ADcoupledAux in production. If they don't overlap do you prefer creating a new object in test/ over templating in the framework ?

Reply:
I would prefer that yea. Then we don't need a bunch of `raw_value` calls